cms/djangoapps/contentstore/courseware_index.py

```diff
@@ -175,8 +181,31 @@ def index_item(item, skip_index=False, groups_usage_info=None):
                 return
 
             item_content_groups = None
+
+            if item.category == "split_test":
+                user_partitions = item.user_partitions
+                split_partitions = []
+                if user_partitions:
+                    for user_partition in user_partitions:
+                        if user_partition.scheme.name == RANDOM_SCHEME:
+                            split_partitions.append(user_partition)
+                for vertical in item.get_children():
+                    if split_partitions:
+                        for user_partition in split_partitions:
+                            for group in user_partition.groups:
+                                group_id = unicode(group.id)
+                                child_location = item.group_id_to_child.get(group_id, None)
```

Comment:
I don't like accessing this internal ``group_id_to_child`` dict directly. Can you refactor some of the methods in ``split_test_module.py`` to give you what you need directly? The closest one I can see is:

    item.description.active_and_inactive_children()

but that's a little more expensive than you need. I'd just introduce a new method to return the child for a given group_id.